src/plonk_util.rs

```diff
@@ -216,3 +229,101 @@ pub fn permutation_polynomial<F: Field>(
     }
     plonk_z_points
 }
+
+pub fn sigma_polynomials<F: Field>(
+    sigma: Vec<usize>,
+    degree: usize,
+    subgroup_generator: F,
+) -> Vec<Vec<F>> {
+    sigma
+        .chunks(degree)
+        .map(|chunk| {
+            chunk
+                .iter()
```

Comment:
Could you switch this to par_iter()? I'm guessing the field exps below would take a nontrivial amount of time, but probably small enough if it's parallelized.